### ISSUE TITLE
BUG: Deleted masks seems to be reactivated when receiving new mails

### ISSUE THREAD

**Issue by Kurpanik** [mozilla/fx-private-relay#3065]
Wir Firefox Relay Premium:

1) Create email mask
2) Delete email mask
3) Receive email on this mask
4) Get spammed because deleted mask reappears and email will still be delivered

This way you don't need Firefox relay

**Comment by Vinnl** [comment]
Hi @Kurpanik, is this a custom mask (e.g. `@<yoursubdomain>.mozmail.com`)? If so, the idea behind those are to be able to come up with an arbitrary address on the spot, e.g. when you're in a physical store. To support that use case, they have to be created on-the-fly when they receive emails.

If you no longer want to receive emails sent to that mask, you can [set it to block all emails](https://support.mozilla.org/en-US/kb/disable-email-forwarding-stop-receiving-emails-sent-through-masks#w_block-email-forwarding-to-your-inbox).

**Comment by Kurpanik** [comment]
Yes, indeed, its a mask from my premium custom domain. And I assumed these custom masks wouldn't behave differently than the normal ones. Just with a custom domain. Well, that's a double-edged sword from a usability point of view, if you can't simply remove such masks from the GUI when someone spams them - but you don't need them anymore, because they were generated for one specific use case. But I see the advantage as well.
